spinalcordtoolbox/scripts/sct_detect_compression.py

```diff
@@ -0,0 +1,296 @@
+                # and `,` is used to convert int back to tuple
+                metrics_agg_merged[key]['Torsion'] = 1/6 * (abs(metrics_agg_merged[key]['MEAN(orientation)'] -
+                                                                metrics_agg_merged[key[0] - 1,]['MEAN(orientation)']) +
+                                                            abs(metrics_agg_merged[key]['MEAN(orientation)'] -
+                                                                metrics_agg_merged[key[0] + 1,]['MEAN(orientation)']) +
+                                                            abs(metrics_agg_merged[key[0] - 1,]['MEAN(orientation)'] -
+                                                                metrics_agg_merged[key[0] - 2,]['MEAN(orientation)']) +
+                                                            abs(metrics_agg_merged[key[0] + 1,]['MEAN(orientation)'] -
+                                                                metrics_agg_merged[key[0] + 2,]['MEAN(orientation)']) +
+                                                            abs(metrics_agg_merged[key[0] - 2,]['MEAN(orientation)'] -
+                                                                metrics_agg_merged[key[0] - 3,]['MEAN(orientation)']) +
+                                                            abs(metrics_agg_merged[key[0] + 2,]['MEAN(orientation)'] -
+                                                                metrics_agg_merged[key[0] + 3,]['MEAN(orientation)']))
+            except Exception as e:
+                # TODO: the warning below is raise mainly in cases of no SC segmentation, for example above the C1 level
+                #  --> consider processing only slices with SC segmentation
+                printv(f"WARNING: Failed to compute torsion for slice {key[0]}. Use `-v 2` flag to see details",
+                       verbose, type='warning')
+                if verbose == 2:
+                    printv(f"\t{e}", verbose)
+                metrics_agg_merged[key]['Torsion'] = None
+        else:
+            metrics_agg_merged[key]['Torsion'] = None
+
+    return metrics_agg_merged
+
+def process_compression(metrics_agg_merged, disc_slices):
+    """
+    Process shape metrics for axial slices at the level of each disc to predict the compression probability.
+    :param metrics_agg_merged: dictionary with aggregated metrics
+    :param disc_slices: dictionary with disc labels as keys and corresponding axial slice numbers as values
+    :return: dataframe with compression probability
+    """
+    compression_df = pd.DataFrame()
+    # Loop across discs
+    for disc, slice in disc_slices.items():
+    # for disc, slices in disc_slices.items():
+    #     # Add one slice above and below to process multiple slices around the disc to compensate for potential
+    #     # disc label shift in the superior-inferior direction
+    #     slices = [slices - 1, slices, slices + 1]
+    #     for slice in slices:
```

Comment:
During debugging, I tried to include more than one slice for each disc.

Because as mentioned in the script description:

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/ff2fde2eb99a55d3d887578200631dda4e14859b/spinalcordtoolbox/scripts/sct_detect_compression.py#L4-L5

The output then looks like this (i.e., the compression probability is predicted for three slices for each disc):
```bash
Disc 6 at axial slice 118 has a high probability of compression: 99.37%. 
Disc 6 at axial slice 119 has a high probability of compression: 99.05%. 
Disc 6 at axial slice 120 has a high probability of compression: 93.22%. 
```

<img width="700" alt="image" src="https://github.com/user-attachments/assets/1af81a8e-b73c-4be5-afca-df035fcdc283" />